pkg/store/cas/cas.go

```diff
@@ -36,24 +40,34 @@ func New(provider ariesstorage.Provider) (*CAS, error) {
 // Write writes the given content to the underlying storage provider.
 // Returns the address of the content.
 func (p *CAS) Write(content []byte) (string, error) {
-	// TODO #318 figure out why the CIDs produced here differ from the ones that IPFS generates.
-	prefix := cid.Prefix{
-		Version:  0,
-		MhType:   mh.SHA2_256,
-		MhLength: -1, // default length
-	}
+	var cid string
+	if p.useIPFSCompatibleV0CIDs {
+		// The v0 CID produced below is equal to what an IPFS node would produce, assuming that:
+		// 1. The IPFS node is running with default settings, and
+		// 2. The size of the content passed in here is less than 256KB (the default chunk size).
+		// Two levels of wrapping are needed. First, the raw data is wrapped as a protobuf UnixFS file, then that needs
+		// to be further wrapped as a protobuf DAG node.
+		cid = merkledag.NodeWithData(unixfs.FilePBData(content, uint64(len(content)))).Cid().String()
+	} else {
+		prefix := gocid.Prefix{
+			Version:  0,
```

Comment:
Why version 0

Reply:
1. That's what it was set to before.
2. That's what our current IPFS client does. I need to update it in another PR to start using v1 as well...

Reply:
(changing it to v1 will break the current sample CIDs)

Reply:
The current sample CIDs seem
incorrect, being neither an accepted v0 nor a v1.

Reply:
What are/ where are the current sample CIDs? Maybe they were produced using the current (faulty) v0 generation in Orb?

In a previous PR I updated the CID in a sample create activity in our BDD tests, but that was to match the CID output by IPFS (by default), so it should be valid. https://github.com/trustbloc/orb/pull/339/files#r626128523